docs/guides/server/hostname2.adoc

```diff
@@ -0,0 +1,137 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Configuring the hostname (v2)"
+summary="Learn how to configure the frontend and backchannel endpoints exposed by {project_name}."
+includedOptions="hostname hostname-* proxy"
+deniedCategories="hostname_v1">
```

Comment:
Actually, one more thing. I think we need to put `hostname_v2` in the denied categories in the old hostname guide.

We should also rename to old hostname guide to include "(v1)" at the end and add some deprecation note in the beginning.